index.ts

```diff
@@ -603,11 +714,589 @@ class ServerlessCustomDomain {
         }
     }
 
+    /**
+     * Determines whether this plug-in is enabled for a websocket endpoint
+     *
+     * This method reads the customDomain.websockets property "enabled" to see if this plug-in should be enabled.
+     * If the property's value is undefined, a default value of true is assumed (for backwards
+     * compatibility).
+     * If the property's value is provided, this should be boolean, otherwise an exception is thrown.
+     * If no customDomain.websockets object exists, websocket creation is disabled.
+     */
+    public evaluateEnabledWs(): boolean {
+        if (typeof this.serverless.service.custom === "undefined"
+            || typeof this.serverless.service.custom.customDomain === "undefined") {
+            throw new Error("serverless-domain-manager: Plugin configuration is missing.");
+        }
+
+        // disable websocket domain creation if configuration missing
+        if (typeof this.serverless.service.custom.customDomain.websockets === "undefined"
+            || this.serverless.service.custom.customDomain.websockets == null) {
+            return false;
+        }
+
+        const enabled = this.serverless.service.custom.customDomain.websockets.enabled;
+
+        if (enabled === undefined) {
+            return true;
+        }
+        if (typeof enabled === "boolean") {
+            return enabled;
+        } else if (typeof enabled === "string" && enabled === "true") {
+            return true;
+        } else if (typeof enabled === "string" && enabled === "false") {
+            return false;
+        }
+        throw new Error(`serverless-domain-manager: Ambiguous enablement boolean: "${enabled}"`);
+    }
+
+    /**
+     * Returns information about custom domain name registered in ApiGatewayV2
+     */
+    public async getDomainInfoWs(): Promise<DomainInfoWs> {
```

Comment:
Same thing in this function as above. 

Maybe try consider having the actual API call to ApiGatewayV2 be abstracted out? i.e. have a function that takes in the endpoint type [rest vs websocket] and possibly the AWS resource [apigateway vs apigatewayv2], and function call [getDomainName]. It would cut down on the duplication of functions that largely did the same thing, just made different api calls.